lib/ruby-debug-ide/xml_printer.rb

```diff
@@ -380,15 +386,46 @@ def max_compact_name_size
       50
     end
 
+    def inspect_with_allocation_control(slice)
+      x = Thread.current
+      
+      start_alloc_size = ObjectSpace.memsize_of_all
+      
+      trace = TracePoint.new(:c_call, :call) do |tp|
+        curr_alloc_size = ObjectSpace.memsize_of_all
+        
+        if(curr_alloc_size - start_alloc_size > 1e7)
```

Comment:
I think the constant here deserves to be put in some constant in the beginning. (even better thing will be to provide the possibility to define such constant via some RUBY_OPT)